### PR TITLE
Changes to Quests, some handling of objects and everything related to The Knight's Sword

### PR DESCRIPTION
**Pull request by KevinSenez**
## What has been done?
- New Quest: The Knight's Sword.
- Changes to Quest Overview interface to handle more data.
- Changes to the Quests system to allow usage of player class (For quests with items to collect, etc.)
- Changes to the Quests Objectives system to allow more versatility.
- Handling of multiple doors, ladders, stairs and a cupboard all relating to the Knights Sword quest. (Around varrock, asgarnia & falador)
- Added npc spawns to regions: 11825, 11925, 12181, thanks to @alycii 
- Added the ability to mine Blurite; as needed for The Knight's Sword.
- Added the ability to purchase the Smithing cape from Thurgo.

## Has your code been documented?
- A little bit yes, will go back at a future time to document everything better.

## What's left to be done?
- Adding definitions to all these new monster spawns.

### CODE REVIEW

File: game/plugins/src/main/kotlin/gg/rsmod/plugins/content/quests/impl/TheKnightsSword.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            }
+        }
+
+        return strings
+    }
+
+    override fun finishQuest(player: Player) {
+        player.advanceToNextStage(this)
+        player.setVarp(QUEST_POINT_VARP, player.getVarp(QUEST_POINT_VARP).plus(pointReward))
+        player.addXp(skill = Skills.MINING, xp = 12725.0)
```
Comment (alycii) [inline game/plugins/src/main/kotlin/gg/rsmod/plugins/content/quests/impl/TheKnightsSword.kt:142]: Shouldn't this be smithing? :p

Comment (KevinSenez) [reply game/plugins/src/main/kotlin/gg/rsmod/plugins/content/quests/impl/TheKnightsSword.kt:142]: Ah shit 😂

**Comment by KevinSenez** [comment]
Issue Fixed, can merge when ready @alycii

**Comment by KevinSenez** [comment]
Added some more changes to go with this PR, also added documentation to my additions.

**Review by alycii — APPROVED** [review]

**Pull request merged**